xamarin/Xamarin.Forms
# Gtk fix transparencies

### Description of Change ###

This change fixes using Transparent color as BakcgroundColor in BoxView and ButtonView elements and adds a new TransparentEventBox to be used as base class for all elements using currenyl EventBox as the base element to draw.

### Issues Resolved ### 

### API Changes ###

No API changes

### Platforms Affected ### 
- GTK 

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 
Before
<img width="807" alt="screen shot 2018-09-07 at 13 28 50" src="https://user-images.githubusercontent.com/111134/45236793-2f8bae80-b2dd-11e8-89d8-208a07246643.png">

After
<img width="804" alt="screen shot 2018-09-07 at 13 36 59" src="https://user-images.githubusercontent.com/111134/45236797-34506280-b2dd-11e8-9814-1a14c11260b7.png">

### Testing Procedure ###
Use the following XAML view to check that elements with Transparent BackgroundColor now draw themselves correctly.
```
<?xml version="1.0" encoding="UTF-8"?>
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms" xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml" x:Class="Samples.Views.TransparencyBug">
    <ContentPage.Content>
        <StackLayout BackgroundColor="Blue" Spacing="10" Padding="10,10,10,10">
            <BoxView HeightRequest="50" BackgroundColor="Transparent"/>
            <BoxView HeightRequest="50" BackgroundColor="Aqua"/>
            <Label HeightRequest="50" Text="This is a label" BackgroundColor="Transparent"/>
            <Label HeightRequest="50" Text="This is a label" BackgroundColor="Aqua"/>
            <Button Padding="5,5,5,5" Margin="5,5,5,5" HeightRequest="50" Text="This is a button" BackgroundColor="Transparent"/>
              <Button Padding="5,5,5,5" Margin="5,5,5,5" HeightRequest="50" Text="This is a button" BackgroundColor="Aqua"/>
            <ListView HeightRequest="50" BackgroundColor="Transparent"/>
        </StackLayout>
    </ContentPage.Content>
</ContentPage>
```

### PR Checklist ###
